e2e/test/iothub/DeviceClientX509AuthenticationE2ETests.cs

```diff
@@ -193,11 +193,11 @@ private async Task X509InvalidDeviceIdOpenAsyncTest(IotHubClientTransportSetting
                 await deviceClient.OpenAsync(cts.Token).ConfigureAwait(false);
                 Assert.Fail("Should throw UnauthorizedException but didn't.");
             }
-            catch (UnauthorizedException)
+            catch (IotHubClientException ex) when (ex.StatusCode is IotHubStatusCode.Unauthorized)
             {
                 // It should always throw UnauthorizedException
             }
-            catch (IotHubCommunicationException ex) when (ex.InnerException is TaskCanceledException)
+            catch (IotHubClientException ex) when (ex.StatusCode is IotHubStatusCode.NetworkErrors && ex.InnerException is TaskCanceledException)
```

Comment:
We should work to not throw `IotHubClientException` when the underlying cause is a signaled cancellation token. Ideas on how we can do that (in a future PR)?

Reply:
I will re-work this in another PR